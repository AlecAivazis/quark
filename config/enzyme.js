// external imports
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import browserEnv from 'browser-env'

// add a request animation frame polyfill
import raf from 'raf'
raf.polyfill()

// enable the react 16 adapter for enzyme
Enzyme.configure({ adapter: new Adapter() })

// add various browser mocks
browserEnv()
