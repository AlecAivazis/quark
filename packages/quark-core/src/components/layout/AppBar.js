// @flow
// external imports
import React from 'react'
import {
    View,
    StyleSheet,
    Text,
    TouchableWithoutFeedback,
    ViewProperties
} from 'react-native'
// local imports
import { baseDim, grey2 } from 'quark-core/styles'

const AppBar = ({
    style,
    statusBarStyle,
    onPress,
    ...unused
}: ViewProperties) => <View style={[styles.container, style]} {...unused} />

const styles = StyleSheet.create({
    container: {
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'flex-start',
        height: 16 * baseDim,
        width: '100%',
        borderBottomWidth: 1,
        borderBottomColor: grey2,
        borderStyle: 'solid',
        paddingLeft: 5 * baseDim,
        paddingRight: 5 * baseDim
    }
})

export default AppBar
