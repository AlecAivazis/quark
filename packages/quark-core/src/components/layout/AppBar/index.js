// @flow
// external imports
import React from 'react'
import { View, StyleSheet, Text, TouchableWithoutFeedback } from 'react-native'
// local imports
import { baseDim, grey2 } from 'quark-core/styles'
import { FlexRow } from '../FlexRow'
import type { FlexViewPropTypes } from '..'

const AppBar = ({ style, statusBarStyle, onPress, ...unused }: FlexViewPropTypes) => (
    <FlexRow
        justifyContent="flex-start"
        alignItems="center"
        style={[styles.container, style]}
        {...unused}
    />
)

const styles = StyleSheet.create({
    container: {
        height: 16 * baseDim,
        width: '100%',
        borderBottomWidth: 1,
        borderBottomColor: grey2,
        borderStyle: 'solid',
        paddingLeft: 5 * baseDim,
        paddingRight: 5 * baseDim
    }
})

export default AppBar
