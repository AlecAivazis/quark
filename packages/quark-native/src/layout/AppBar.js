// external imports
import React from 'react'
import { View, StyleSheet, Text } from 'react-native'
// local imports
import { StatusBar } from '.'
import { baseDim, grey2 } from 'quark-core/styles'

const AppBar = ({style, statusBarStyle, ...unused}) => (
    <View style={styles.container}>
        <StatusBar style={statusBarStyle}/>
        <View style={[styles.content, style]} {...unused}/>
    </View>
)

const styles = StyleSheet.create({
    container: {
        width: '100%',
        borderBottomWidth: 1,
        borderBottomColor: grey2,
        borderStyle: 'solid',
    },
    content: {
        display: 'flex',
        height: 16 * baseDim,
        paddingLeft: 5 * baseDim,
        paddingRight: 5 * baseDim,
        justifyContent: 'center',
        alignItems: 'flex-start',
    },
})

export default AppBar
