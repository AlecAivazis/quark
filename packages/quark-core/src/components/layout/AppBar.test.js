// external imports
import React from 'react'
import { Text } from 'react-native'
import { mount } from 'enzyme'
// local imports
import AppBar from './AppBar'

it('renders', () => {
    const rendered = mount(
        <AppBar>
            <Text>hello</Text>
        </AppBar>
    )

    expect(rendered).toMatchSnapshot()
})
