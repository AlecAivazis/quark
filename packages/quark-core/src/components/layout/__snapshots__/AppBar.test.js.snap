// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<AppBar>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "borderBottomColor": "#EAEAEE",
        "borderBottomWidth": 1,
        "borderStyle": "solid",
        "display": "flex",
        "height": 64,
        "justifyContent": "center",
        "paddingLeft": 20,
        "paddingRight": 20,
        "width": "100%",
      }
    }
  >
    <div
      className="rn-alignItems-1habvwh rn-borderBottomColor-156pb2y rn-borderBottomWidth-qklmqi rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-height-sga3zk rn-justifyContent-1777fci rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingBottom-1mdbw0j rn-paddingLeft-b5h31w rn-paddingRight-1ah4tor rn-position-bnwqim rn-width-13qz1uu rn-zIndex-1lgpqti"
    >
      <Text>
        <div
          className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1b43r93 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
          dir="auto"
        >
          hello
        </div>
      </Text>
    </div>
  </View>
</AppBar>
`;
